gh-110590: Fix `_sre.compile` error overwrite (#110591)
python/cpython: Misc/NEWS.d/next/Library/2023-10-10-10-46-55.gh-issue-110590.fatz-h.rst

Fix ``_sre.compile`` error overwriting `TypeError` with `OverflowError` when

> ``code`` arguments was a list of non-ints.

Reviewer: Could you please add proper Sphinx markup here?

> Fix ``_sre.compile`` error overwriting ``TypeError``
> with ``OverflowError`` when ``code`` argument
> was a list of non-ints.

Reviewer: ```suggestion
Fix a bug in :meth:`!_sre.compile` where :exc:`TypeError`
would be overwritten by :exc:`OverflowError` when
the *code* argument was a list of non-ints.
```
